src/clojure/nrepl/cmdline.clj

```diff
@@ -440,16 +438,15 @@ Exit:      Control+D or (exit) or (quit)"
   connection details from.
   Takes nREPL server map and processed CLI options map.
   Returns connection header string."
-  [server options]
-  (let [transport (:transport options)
-        ^java.net.ServerSocket ssocket (:server-socket server)
-        ^URI uri (socket/as-nrepl-uri ssocket (transport/uri-scheme transport))]
-    ;; The format here is important, as some tools (e.g. CIDER) parse the string
-    ;; to extract from it the host and the port to connect to
-    (if-let [host (.getHost uri)]
+  [{:keys [host port socket] :as server}
+   {:keys [transport]}]
+  (let [uri (socket/as-nrepl-uri server (transport/uri-scheme transport))]
+    ;; The format here is important, as some tools (e.g. CIDER) parse the
+    ;; string to extract from it the host and the port to connect to
+    (if socket
+      (str "nREPL server started on socket " (.toASCIIString uri))
```

Comment:
Why is this preferable over `.getHost`? We want to avoid unicode hostnames or what?